tests/e2e/csi_snapshot_basic.go

```diff
@@ -195,4 +194,315 @@ var _ = ginkgo.Describe("[block-vanilla-snapshot] Volume Snapshot Basic Test", f
 		err = verifySnapshotIsDeletedInCNS(volHandle, snapshotId)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
+
+	/*
+		Create/Delete snapshot via k8s API using VolumeSnapshotContent (Pre-Provisioned Snapshots)
+
+		1. Create a storage class (eg: vsan default) and create a pvc using this sc
+		2. The volumesnapshotclass is set to delete
+		3. Create a VolumeSnapshotContent using snapshot-handle
+		   a. get snapshotHandle by referring to an existing volume snapshot
+		   b. this snapshot will be created dynamically, and the snapshot-content that is
+		      created by that will be referred to get the snapshotHandle
+		4. Create a volume snapshot using source set to volumeSnapshotContentName above
```

Comment:
For the dynamically provisioned snapshot, deletion policy is Delete.
Create the 2nd VolumeSnapshotContent with deletion policy set to Retain.
The order of deletion will be:
- Delete the 2nd VolumeSnapshot, the 2nd VolumeSnapshotContent along with CNS snapshot will not be deleted.
- Delete the 1st VolumeSnapshot, the 1st VolumeSnapshotContent along with CNS snapshot will be deleted.
- Delete the 2nd VolumeSnapshotContent.

Reply:
Xing, as discussed offline - I have added a new test with the steps given in the comment